packages/block-library/src/gallery/index.php

```diff
@@ -32,12 +32,53 @@ function block_core_gallery_data_id_backcompatibility( $parsed_block ) {
 
 add_filter( 'render_block_data', 'block_core_gallery_data_id_backcompatibility' );
 
+/**
+ * Adds a style tag for the --wp--style--unstable-gallery-gap var.
+ *
+ * The Gallery block needs to recalculate Image block width based on
+ * the current gap setting in order to maintain the number of flex columns
+ * so a css var is added to allow this.
+ *
+ * @param array  $attributes Attributes of the block being rendered.
+ * @param string $content Content of the block being rendered.
+ * @return string The content of the block being rendered.
+ */
+function block_core_gallery_render( $attributes, $content ) {
+	$gap = _wp_array_get( $attributes, array( 'style', 'spacing', 'blockGap' ) );
+	// Skip if gap value contains unsupported characters.
+	// Regex for CSS value borrowed from `safecss_filter_attr`, and used here
+	// because we only want to match against the value, not the CSS attribute.
+	$gap       = preg_match( '%[\\\(&=}]|/\*%', $gap ) ? null : $gap;
+	$id        = uniqid();
+	$class     = 'wp-block-gallery-' . $id;
+	$content   = preg_replace(
+		'/' . preg_quote( 'class="', '/' ) . '/',
```

Comment:
Curious why we have this `preg_quote` here as it doesn't do anything. We generally never need to RegExp-quote a static string literal, especially if the string value doesn't have the delimiter we're escaping in it, as this one does.

Note too that this PCRE pattern isn't doing anything that simple string substitution doesn't do already - why did we use a `preg_replace()` here instead of `str_replace()`?

Reply:
I'm not too sure why it was used originally but I believe the implementation here was duplicated from `layout.php` as it matches the logic used here: https://github.com/WordPress/gutenberg/blob/trunk/lib/block-supports/layout.php#L318, which doesn't appear to have been updated from when the Layout block support was initially introduced back in: https://github.com/WordPress/gutenberg/pull/29335, unless I'm missing something.

It sound like `str_replace()` would work too and be simpler to read. If we update it here it'd be good to update it in the Layout support too for consistency. And of course I'm hopeful for us to be able to switch over to [WP_HTML_Walker](https://github.com/WordPress/gutenberg/pull/42485) at some point 😀

Reply:
Yes, it was definitely a case of copy pasta from layout, so @youknowriad is potentially the one that knows the reason for the use of `preg_replace` if there was one.

Reply:
Something like `$content = str_replace( 'wp-block-gallery', "wp-block-gallery $class", $content );` seems to work in this instance. There are several nested `<figure class="...`, hence the use of the classname. 

Doing a dodgy `microtime` speed test with 10 galleries on a page, the difference appears negligible on my machine.

Reply:
> Doing a dodgy microtime speed test with 10 galleries on a page, the difference appears negligible on my machine.

My question was definitely geared towards "this looks obviously wrong so I'm confident that I'm misunderstanding the intent of the code and its implementation" more than "this could potentially be slower than another micro-optimized approach."

Thanks for the clarification y'all. In the meantime, would someone be willing to simplify so this doesn't confuse even more folks? To be clear, I'm not commenting on the method of HTML modification, just on the superfluous use of `preg_` functions which emulate `str_` varieties.

Well, it may be the case that `str_replace()` won't _only_ replace a single instance. If that's important we might need to stick with `preg_` but we absolutely don't need `preg_quote()` because it's being passed a literal with no quotable characters.

```php
$content = preg_replace( '/class="/', 'class="' . $class . ' ', $content, 1 );
```

> And of course I'm hopeful for us to be able to switch over to https://github.com/WordPress/gutenberg/pull/42485 at some point 😀

Yes indeed

```php
$w = new WP_HTML_Walker( $html );
$w->next_tag( [ 'class_name' => "wp-block-gallery-{$id}" ] );
$w->add_class( $class );
$content = (string) $w;
```